apis/v1alpha1/gateway_types.go

```diff
@@ -483,47 +485,125 @@ type GatewayStatus struct {
 	//
 	// These addresses should all be of type "IPAddress".
 	//
-	// +required
+	// +optional
 	Addresses []GatewayAddress `json:"addresses"`
 
 	// Conditions describe the current conditions of the Gateway.
-	// +optional
+	//
+	// Implementations should prefer to express Gateway conditions
+	// using the `GatewayConditionType` and `GatewayConditionReason`
+	// constants so that operators and tools can converge on a common
+	// vocabulary to describe Gateway state.
+	//
+	// Known condition types are:
+	//
+	// * "Scheduled"
+	// * "Ready"
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
```

Comment:
Nice details here, we should specify these in more places. It looks like the `+patchMergeKey` and `+patchStrategy` labels are not being translated into anything meaningful for the CRD spec though? Maybe we should only rely on the `+listType` and `+listMapKey` type.

Reply:
> Nice details here, we should specify these in more places. It looks like the `+patchMergeKey` and `+patchStrategy` labels are not being translated into anything meaningful for the CRD spec though? Maybe we should only rely on the `+listType` and `+listMapKey` type.

Huh, maybe we need a different CRD generator?

Reply:
Are field tags required to specify the patch strategy? See https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1@v0.19.0?tab=doc#Condition and https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/342